google/cloud/firestore_v1/async_transaction.py

```diff
@@ -182,18 +188,30 @@ async def get(
                 should be retried.  Defaults to a system-specified policy.
             timeout (float): The timeout for this request.  Defaults to a
                 system-specified value.
+            explain_options
+                (Optional[:class:`~google.cloud.firestore_v1.query_profile.ExplainOptions`]):
+                Options to enable query profiling for this query. When set,
+                explain_metrics will be available on the returned generator.
 
         Yields:
-            .DocumentSnapshot: The next document snapshot that fulfills the
-            query, or :data:`None` if the document does not exist.
+            DocumentSnapshot: The next document snapshot that fulfills the query,
+            or :data:`None` if the document does not exist.
         """
         kwargs = _helpers.make_retry_timeout_kwargs(retry, timeout)
         if isinstance(ref_or_query, AsyncDocumentReference):
+            if explain_options is not None:
+                warnings.warn(
+                    "explain_options not supported in transanction with "
+                    "document references and will be ignored. To use "
+                    "explain_options, use transaction with query instead."
+                )
```

Comment:
Maybe good to reference the `ref_or_query` argument name here, so they know what to change?

I'm still kind of considering whether it makes sense to warn or raise an exception here. I'm curious if you have strong thoughts one way or the other

Reply:
Originally I only added a warning because I didn't want to block people's workflows. Now that I contemplate it, if we raise an exception, it's a little less convenient for the user but perhaps it's less error prone? For example if someone sets`explain_options(analyze=False)` for document references, expecting the query to not run (thus not spending money) but then they receive a warning and the query is run. Or in some situations warnings are completely filtered out so people never see it. Maybe we should raise exception to prevent situations like this. WDYT?

Reply:
I think we don't need to worry about breaking workflows here, since explain_options is a new flag

I'm a bit split on it, but I think I'd lean towards using an exception here, and then we can always loosen it to a warning if we get feedback that it makes certain things difficult.

You should add a mention of this limitation in the `explain_options` section of the docstring (and maybe `ref_or_query`?), and add a `Throws` section if you go with the exception

Reply:
Regardless I will add type hints for `ref_or_query`, and comments saying `explain_options` doesn't apply to documents.

Reply:
I did the following:
- change warning to exception
- type hint for `ref_or_query`
- note restriction in `explain_options` section
- add `Raises` section